crates/worker/src/partition/state_machine/mod.rs

```diff
@@ -1064,16 +1105,269 @@ mod tests {
             assert_that!(
                 actions,
                 all!(
-                    contains(pat!(Action::IngressResponse(pat!(IngressResponse {
-                        target_node: eq(ingress_id_1),
-                        idempotency_id: some(eq(idempotency_id.clone())),
-                        response: eq(ResponseResult::Success(response_bytes.clone()))
-                    })))),
-                    contains(pat!(Action::IngressResponse(pat!(IngressResponse {
-                        target_node: eq(ingress_id_2),
-                        idempotency_id: some(eq(idempotency_id.clone())),
-                        response: eq(ResponseResult::Success(response_bytes.clone()))
-                    })))),
+                    contains(pat!(Action::IngressResponse(pat!(
+                        IngressResponseEnvelope {
+                            target_node: eq(ingress_id_1),
+                            inner: pat!(ingress::InvocationResponse {
+                                correlation_ids: pat!(ingress::InvocationResponseCorrelationIds {
+                                    idempotency_id: some(eq(idempotency_id.clone())),
+                                }),
+                                response: eq(IngressResponseResult::Success(
+                                    invocation_target.clone(),
+                                    response_bytes.clone()
+                                ))
+                            })
+                        }
+                    )))),
+                    contains(pat!(Action::IngressResponse(pat!(
+                        IngressResponseEnvelope {
+                            target_node: eq(ingress_id_2),
+                            inner: pat!(ingress::InvocationResponse {
+                                correlation_ids: pat!(ingress::InvocationResponseCorrelationIds {
+                                    idempotency_id: some(eq(idempotency_id.clone())),
+                                }),
+                                response: eq(IngressResponseResult::Success(
+                                    invocation_target.clone(),
+                                    response_bytes.clone()
+                                ))
+                            })
+                        }
+                    ))))
+                )
+            );
+        }
+
+        #[test(tokio::test)]
+        async fn latch_invocation_while_executing_with_send() {
```

Comment:
How does latch relate to attach? Isn't this more or less the same? If so, can we unify the terminology?